tests/phpunit/tests/query/searchColumns.php

```diff
@@ -0,0 +1,414 @@
+
+		$this->assertSameSetsWithIndex( array( self::$pid2, self::$pid3 ), $q->posts );
+	}
+
+	/**
+	 * Tests that search supports search columns when using term exclusion.
+	 *
+	 * @ticket 43867
+	 */
+	public function test_s_should_support_search_columns_when_searching_with_term_exclusion() {
+		$q = new WP_Query(
+			array(
+				's'              => 'bar -baz',
+				'search_columns' => array( 'post_excerpt', 'post_content' ),
+				'fields'         => 'ids',
+			)
+		);
+
+		$this->assertSameSetsWithIndex( array( self::$pid2 ), $q->posts );
+	}
+
+	/**
+	 * Tests that search columns is filterable with the `post_search_columns` filter.
+	 *
+	 * @ticket 43867
+	 */
+	public function test_search_columns_should_be_filterable() {
+		add_filter( 'post_search_columns', array( $this, 'post_supported_search_column' ), 10, 3 );
+		$q = new WP_Query(
+			array(
+				's'      => 'foo',
+				'fields' => 'ids',
+			)
+		);
+
+		$this->assertSameSetsWithIndex( array( self::$pid1 ), $q->posts );
+	}
+
+	/**
+	 * Filter callback that sets a supported search column.
+	 *
+	 * @param  string[] $search_columns Array of column names to be searched.
+	 * @param  string   $search         Text being searched.
+	 * @param  WP_Query $wp_query       The current WP_Query instance.
+	 * @return string[] $search_columns Array of column names to be searched.
+	 */
+	public function post_supported_search_column( $search_columns, $search, $wp_query ) {
+		$search_columns = array( 'post_title' );
+		return $search_columns;
+	}
+
+	/**
+	 * Tests that search columns ignores non-supported search columns from the `post_search_columns` filter.
+	 *
+	 * @ticket 43867
+	 */
+	public function test_search_columns_should_not_filterable_when_non_supported_search_columns() {
+		add_filter( 'post_search_columns', array( $this, 'post_non_supported_search_column' ), 10, 3 );
+		$q = new WP_Query(
+			array(
+				's'      => 'foo',
+				'fields' => 'ids',
+			)
+		);
+
+		$this->assertStringNotContainsString( 'post_name', $q->request, "SQL request shouldn't contain post_name string." );
+		$this->assertSameSetsWithIndex( array( self::$pid1, self::$pid2, self::$pid3 ), $q->posts, 'Query results should be equal to the set.' );
+	}
+
+	/**
+	 * Filter callback that sets an existing but non-supported search column.
+	 *
+	 * @param  string[] $search_columns Array of column names to be searched.
+	 * @param  string   $search         Text being searched.
+	 * @param  WP_Query $wp_query       The current WP_Query instance.
+	 * @return string[] $search_columns Array of column names to be searched.
+	 */
+	public function post_non_supported_search_column( $search_columns, $search, $wp_query ) {
+		$search_columns = array( 'post_name' );
+		return $search_columns;
+	}
+
+	/**
+	 * Tests that search columns ignores non-existing search columns from the `post_search_columns` filter.
+	 *
+	 * @ticket 43867
+	 */
+	public function xtest_search_columns_should_not_filterable_non_existing_search_column() {
```

Comment:
```suggestion
	public function test_search_columns_should_not_be_filterable_with_non_existing_search_column() {
```
This also had an `x` at the beginning of the test name (i.e. `xtest_`), which meant the test wasn't running. I removed this locally and the test passes, so I assume this `x` was just a typo and should also be removed.